src/object/srv_obj.c

```diff
@@ -1534,14 +1534,15 @@ ds_obj_rw_handler(crt_rpc_t *rpc)
 
 	/* Stop the distribute transaction */
 	rc = dtx_leader_end(&dlh, ioc.ioc_coc, rc);
-	if (rc == -DER_AGAIN) {
-		if (dlh.dlh_handle.dth_renew) {
-			/* epoch conflict, renew it and retry. */
-			orw->orw_epoch = crt_hlc_get();
-			flags &= ~ORF_RESEND;
-			memset(&dlh, 0, sizeof(dlh));
-			D_GOTO(renew, rc);
-		}
+	if (rc == -DER_TX_RESTART) {
+		D_ASSERT(dlh.dlh_handle.dth_renew);
+		/* epoch conflict, renew it and retry. */
+		orw->orw_epoch = crt_hlc_get();
+		flags &= ~ORF_RESEND;
+		memset(&dlh, 0, sizeof(dlh));
+		D_GOTO(renew, rc);
+	} else if (rc == -DER_AGAIN) {
```

Comment:
I am thinking that whether we still have "-DER_AGAIN" case from non-leader? If yes, can we make the non-leader to retry related modification by itself without replying -DER_AGAIN to the leader?

Reply:
Not sure.  I don’t think VOS returns it but I saw some logic in the DTX code that translates DER_INPROGRESS to DER_AGAIN so I left that logic as is